lib/pbench/test/unit/server/database/test_datasets_metadata_db.py

```diff
@@ -190,41 +189,30 @@ def test_user_metadata(self, attach_dataset):
         ds = Dataset.query(name="drb")
         user1 = User.query(username="drb")
         assert ds.metadatas == []
-        t = Metadata.create(key="user", value=True, dataset=ds, user=user1)
+        t = Metadata.create(key="user", value=True, dataset=ds, user=user1.username)
         assert t is not None
         assert ds.metadatas == [t]
-        assert user1.dataset_metadata == [t]
 
         user2 = User.query(username="test")
-        d = Metadata.create(key="user", value=False, dataset=ds, user=user2)
+        d = Metadata.create(key="user", value=False, dataset=ds, user=user2.username)
         assert d is not None
         assert ds.metadatas == [t, d]
-        assert user1.dataset_metadata == [t]
-        assert user2.dataset_metadata == [d]
```

Comment:
I feel like the loss of the assertions at lines 202-203 is a problem.  Just because we're changing the definition of a user doesn't mean that we're changing the definition of user-associated metadata!  It should still be the case that `user1` sees its own metadata and that it is different from what `user2` sees.

Or am I misunderstanding things, here?

Reply:
These asserts are a consequence of the SQLAlchemy relationship statement. The query resolves to a join, which provides the associated `Metadata` instances as a list on the `User`. We're getting rid of that, so these asserts are no longer possible.

The equivalent would be a new `Metadata.query` (which we don't exactly have) to return all metadata rows with the designated user UUID as owner. This would be good to add. Although as it's not an interface we need to support in the `Metadata` class, having the test code directly execute the query would probably be best. E.g., 

```python
    assert Database.db_session.query(Metadata).filter(owner=uuid1).all() == [t]
```